product_code_mandatory/__init__.py

```diff
@@ -0,0 +1,16 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from . import models
+
+
+def pre_init_product_code(cr):
+    cr.execute("""UPDATE product_template
+        SET default_code = 'DEFAULT' || nextval('ir_default_id_seq')
+        WHERE default_code is NULL
+        OR LENGTH(default_code) = 0""")
+    cr.execute("""UPDATE product_product
```

Comment:
I would update the product.product with a reference from template, let's say the product template has DEFAULT100, and on variants associated you get DEFAULT100-1, DEFAULT100-2, when you search by template code on products, to have the variantss from it...

Reply:
@feketemihai thanks for the comment. 

Let me quote Max' comment:
We don't need to touch product_template as the default_code of a product template with many variants will be empty:

https://github.com/odoo/odoo/blob/11.0/addons/product/models/product_template.py#L256-L266

With Odoo's default behavior when a product.template is created it creates a product.product record in which the product.product.default_code is initialized with a unique value. This value then is pushed back to product.template.default_code.

Reply:
@agyamuta Be careful my comment was referring to the case of uniqueness. In this case, the field cannot be empty.